x-pack/plugins/apm/server/lib/services/get_service_dependencies/get_destination_map.ts

```diff
@@ -93,15 +92,15 @@ export const getDestinationMap = ({
 
     const outgoingConnections =
       response.aggregations?.connections.buckets.map((bucket) => {
-        const doc = bucket.docs.hits.hits[0];
+        const fieldValues = bucket.sample.top[0].metrics;
 
         return {
           [SPAN_DESTINATION_SERVICE_RESOURCE]: String(
             bucket.key[SPAN_DESTINATION_SERVICE_RESOURCE]
           ),
-          [SPAN_ID]: String(doc.fields[SPAN_ID]?.[0]),
-          [SPAN_TYPE]: String(doc.fields[SPAN_TYPE]?.[0] ?? ''),
-          [SPAN_SUBTYPE]: String(doc.fields[SPAN_SUBTYPE]?.[0] ?? ''),
+          [SPAN_ID]: (fieldValues[SPAN_ID] ?? '') as string,
```

Comment:
Is `as string` necessary when you are already have a default value (empty string)

```suggestion
          [SPAN_ID]: (fieldValues[SPAN_ID] ?? '') as string,
```

Reply:
the type is number | string | null (I think). So ?? '' only refines to string | number. I can also cast to a string by calling String(fieldValues[SPAN_ID)), but I'm not sure if that's better, as that is runtime code rather than a type assertion. What do you think?

Reply:
Okay, fine to leave it.